db/manual_compaction_test.cc

```diff
@@ -149,7 +177,121 @@ TEST_F(ManualCompactionTest, Test) {
 
   // close database
   delete db;
-  DestroyDB(dbname_, ROCKSDB_NAMESPACE::Options());
+  DestroyDB(dbname_, Options());
+}
+
+TEST_F(ManualCompactionTest, SkipLevel) {
+  DB* db;
+  Options options;
+  options.num_levels = 3;
+  options.level0_file_num_compaction_trigger = INT_MAX;
+  options.compaction_style = kCompactionStyleLevel;
+  options.create_if_missing = true;
+  options.compression = kNoCompression;
+  LogCompactionFilter* filter = new LogCompactionFilter();
+  options.compaction_filter = filter;
+  ASSERT_OK(DB::Open(options, dbname_, &db));
+
+  // Keys: 1, 2, 4, 8
+  // Each key is flushed to a new SST file in L0
+  for (int k = 1; k <= 8; k <<= 1) {
+    db->Put(WriteOptions(), std::to_string(k), "");
+    db->Flush(FlushOptions());
+  }
+
+  {
+    // L0: 1, 2, 4, 8
+    // [5, 7] has no overlap with any file in any level.
+    Slice start("5");
+    Slice end("7");
+    filter->Reset();
+    db->CompactRange(CompactRangeOptions(), &start, &end);
+    ASSERT_EQ(0, filter->NumKeys());
+  }
+
+  {
+    // L0: 1, 2, 4, 8
+    // [3, 7] overlaps with 4 in L0
```

Comment:
I think we need a test case for something like: L0 file 1, 2, [4, 8] and the compaction range is [3, 7] and we test that compaction doesn't pick L0.

Reply:
in this case, [5, 7] won't pick L0, but [3, 7] will.

Reply:
Right. Maybe test both cases:)

Reply:
updated the SkipLevel test to cover this case